### PR TITLE
Make CompareDofObjectsByID available to all

### PR DESCRIPTION
**Pull request by lindsayad**
This is a useful struct for us in MOOSE mortar

### CODE REVIEW

File: include/base/dof_object.h [hunk 1]
```diff
@@ -1325,6 +1325,31 @@ unsigned int DofObject::system_var_to_vg_var (const unsigned int s,
...
+  bool operator()(const DofObject * a,
+                  const DofObject * b) const
+  {
+    libmesh_assert (a);
+    libmesh_assert (b);
+
+    return a->id() < b->id();
+  }
+
+  bool operator()(DofObject * a,
```
Comment (jwpeterson) [inline include/base/dof_object.h:1344]: I see that you added a version taking non-const inputs, but I don't understand why it's needed. I guess you got a compiler error that caused you to add it?

Comment (lindsayad) [reply include/base/dof_object.h:1344]: I do have a container of pointers to non-const and I preeemptively assumed I would get a compiler error saying that it was unable to convert pointer to non-const to pointer to const ... but I can double-check

Comment (lindsayad) [reply include/base/dof_object.h:1344]: Huh I can't believe I didn't know that you can implicitly convert pointer to non-const to pointer to const. I mean it makes sense but for some reason I thought it was illegal

Comment (lindsayad) [reply include/base/dof_object.h]: Ok I've removed the non-const pointer overload

**Review by jwpeterson — APPROVED** [review]

**Pull request merged**